File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/registruIntrariIesiri/RegistruIesiriViewModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.registruIntrariIesiri;

import java.util.Date;
import java.util.List;

import ro.cloudSoft.cloudDoc.domain.AtasamentModel;

public class RegistruIesiriViewModel {

	private Long id;
	private String numarInregistrare;
	private Date dataInregistrare;
	private String tipDocument;
	private String codTipDocument;
	private Integer numarPagini;
	private Integer numarAnexe;
	private String intocmitDeUser;
	private boolean trimisPeMail;
	private String continut;
	private boolean asteptamRaspuns;
	private Date termenRaspuns;
	private List<RegistruIesiriDestinatarViewModel> destinatari;
	private String numeProiecteConcatenate;
	private boolean anulat;
	private String motivAnulare;
	private boolean inchis;
	private boolean trebuieAnulat;
	private List<AtasamentModel> atasamente;
	private String subactivityName;
	
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public String getNumarInregistrare() {
		return numarInregistrare;
	}
	public void setNumarInregistrare(String numarInregistrare) {
		this.numarInregistrare = numarInregistrare;
	}
	public Date getDataInregistrare() {
		return dataInregistrare;
	}
	public void setDataInregistrare(Date dataInregistrare) {
		this.dataInregistrare = dataInregistrare;
	}
	public String getTipDocument() {
		return tipDocument;
	}
	public void setTipDocument(String tipDocument) {
		this.tipDocument = tipDocument;
	}
	public String getCodTipDocument() {
		return codTipDocument;
	}
	public void setCodTipDocument(String codTipDocument) {
		this.codTipDocument = codTipDocument;
	}
	public Integer getNumarPagini() {
		return numarPagini;
	}
	public void setNumarPagini(Integer numarPagini) {
		this.numarPagini = numarPagini;
	}
	public Integer getNumarAnexe() {
		return numarAnexe;
	}
	public void setNumarAnexe(Integer numarAnexe) {
		this.numarAnexe = numarAnexe;
	}
	public String getIntocmitDeUser() {
		return intocmitDeUser;
	}
	public void setIntocmitDeUser(String intocmitDeUser) {
		this.intocmitDeUser = intocmitDeUser;
	}
	public boolean isTrimisPeMail() {
		return trimisPeMail;
	}
	public void setTrimisPeMail(boolean trimisPeMail) {
		this.trimisPeMail = trimisPeMail;
	}
	public String getContinut() {
		return continut;
	}
	public void setContinut(String continut) {
		this.continut = continut;
	}
	public boolean isAsteptamRaspuns() {
		return asteptamRaspuns;
	}
	public void setAsteptamRaspuns(boolean asteptamRaspuns) {
		this.asteptamRaspuns = asteptamRaspuns;
	}
	public Date getTermenRaspuns() {
		return termenRaspuns;
	}
	public void setTermenRaspuns(Date termenRaspuns) {
		this.termenRaspuns = termenRaspuns;
	}
	public List<RegistruIesiriDestinatarViewModel> getDestinatari() {
		return destinatari;
	}
	public void setDestinatari(List<RegistruIesiriDestinatarViewModel> destinatari) {
		this.destinatari = destinatari;
	}
	public String getNumeProiecteConcatenate() {
		return numeProiecteConcatenate;
	}
	public void setNumeProiecteConcatenate(String numeProiecteConcatenate) {
		this.numeProiecteConcatenate = numeProiecteConcatenate;
	}
	public boolean isAnulat() {
		return anulat;
	}
	public void setAnulat(boolean anulat) {
		this.anulat = anulat;
	}
	public String getMotivAnulare() {
		return motivAnulare;
	}
	public void setMotivAnulare(String motivAnulare) {
		this.motivAnulare = motivAnulare;
	}
	public boolean isInchis() {
		return inchis;
	}
	public void setInchis(boolean inchis) {
		this.inchis = inchis;
	}
	public List<AtasamentModel> getAtasamente() {
		return atasamente;
	}
	public void setAtasamente(List<AtasamentModel> atasamente) {
		this.atasamente = atasamente;
	}
	public boolean isTrebuieAnulat() {
		return trebuieAnulat;
	}
	public void setTrebuieAnulat(boolean trebuieAnulat) {
		this.trebuieAnulat = trebuieAnulat;
	}
	public String getSubactivityName() {
		return subactivityName;
	}
	public void setSubactivityName(String subactivityName) {
		this.subactivityName = subactivityName;
	}
	
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/registru-intrari-iesiri/registru-iesiri-view.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { RegistruIesiriDestinatarViewModel } from "./registru-iesiri-destinatari-view.model";
import { JsonDateConverter } from "../../json-mapper";
import { DateUtils } from "../../utils";
import { DateConstants } from "../../constants";
import { RegistruIesiriAtasamentModel } from "./registru-iesiri-atasamente.model";

@JsonObject
export class RegistruIesiriViewModel {

	@JsonProperty("id", Number)
	public id: number = null;

	@JsonProperty("numarInregistrare", String)
	public numarInregistrare: string = null;

	@JsonProperty("dataInregistrare", JsonDateConverter)
	public dataInregistrare: Date = null;

	@JsonProperty("tipDocument", String)
	public tipDocument: string = null;

	@JsonProperty("codTipDocument", String)
	public codTipDocument: string = null;

	@JsonProperty("numarPagini", Number)
	public numarPagini: number = null;

	@JsonProperty("numarAnexe", Number)
	public numarAnexe: number = null;

	@JsonProperty("intocmitDeUser", String)
	public intocmitDeUser: string = null;

	@JsonProperty("trimisPeMail", Boolean)
	public trimisPeMail: boolean = null;

	@JsonProperty("continut", String)
	public continut: string = null;

	@JsonProperty("asteptamRaspuns", Boolean)
	public asteptamRaspuns: boolean = null;

	@JsonProperty("termenRaspuns", JsonDateConverter)
	public termenRaspuns: Date = null;

	@JsonProperty("destinatari", [RegistruIesiriDestinatarViewModel])
	public destinatari: RegistruIesiriDestinatarViewModel[] = [];

	@JsonProperty("numeProiecteConcatenate", String)
	public numeProiecteConcatenate: string = null;

	@JsonProperty("anulat", Boolean)
	public anulat: boolean = false;

	@JsonProperty("motivAnulare", String)
	public motivAnulare: string = null;
	
	@JsonProperty("inchis", Boolean)
	public finalizat: boolean = null;

	@JsonProperty("trebuieAnulat", Boolean, true)
	public trebuieAnulat: boolean = null;

	@JsonProperty("subactivityName", String)
	public subactivityName: string = null;

	public get dataInregistrareForDisplay(): string {
		return DateUtils.formatForDisplay(this.dataInregistrare);
	}

	public get trimisPeMailForDisplay(): string {
		if (this.trimisPeMail) {
			return "LABELS.YES";
		}
		return "LABELS.NO";
	}

	public get necesitaRaspunsForDisplay(): string {
		if (this.asteptamRaspuns) {
			return "LABELS.YES";
		}
		return "LABELS.NO";
	}

	public get anulatForDisplay(): string {
		if (this.anulat) {
			return "LABELS.YES";
		}
		return "LABELS.NO";
	}

	public get termenRaspunsForDisplay(): string {
		return DateUtils.formatForDisplay(this.termenRaspuns);
	}

	public get luna(): number {
		return this.dataInregistrare.getMonth();
	}

	public get lunaForDisplay(): string {
		return "LABELS." + DateConstants.getMonthCodeByIndex(this.dataInregistrare.getMonth());
	}

	public get numarInregistrareForDisplay(): string {
		let numarInregistrarePerAn: string = this.numarInregistrare;
		let numarInregistrareSplitArray: string[] = numarInregistrarePerAn.split("~");
		return numarInregistrareSplitArray[0];
	}

	public get numarInregistrareForFilter(): number {
		let numarInregistrarePerAn: string = this.numarInregistrare;
		let numarInregistrareSplitArray: string[] = numarInregistrarePerAn.split("~");

		return Number(numarInregistrareSplitArray[0]);
	}

	public get finalizatForDisplay(): string {
		if (this.finalizat) {
			return "LABELS.YES";
		}
		return "LABELS.NO";
	}
}